RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -40,10 +40,33 @@ public static string ShellCommand(BotData data, string executable, string argume
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
 
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, string scriptOrFile, object[] parameters, bool isScript = false)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript)
+            {
+                // If it's a script, use InvokeFromStringAsync
+                result = await StaticNodeJSService.InvokeFromStringAsync<T>(
+                    scriptOrFile,  // This is now the script content
+                    null,          // newCacheIdentifier
```

Comment:
The whole idea behind using `InvokeFromFileAsync` is that, since this script can be called very frequently by multiple bots, it is better to cache it. According to the [documentation](https://github.com/JeringTech/Javascript.NodeJS), when using `InvokeFromFileAsync`, the file is cached using its absolute path on disk. When using `InvokeFromStringAsync`, we would need to provide a cache key, otherwise it can have an impact on performance.

I think in this case the cache key could be the MD5 of the script's string. Take a look at the repo's docs for a sample usage, trying to invoke from cache first using `TryInvokeFromCacheAsync`, and then on a cache miss using `InvokeFromStringAsync` by passing the cache key so it's cached.